### PR TITLE
baseline: Inline instruction requirements check

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Comment by gumb0** [comment]
```
baseline/analyse/main/blake2b_huff_mean                                     +0.0058         +0.0055            15            15            15            15
baseline/execute/main/blake2b_huff/empty_mean                               +0.0037         +0.0035            44            44            44            44
baseline/total/main/blake2b_huff/empty_mean                                 -0.0087         -0.0089            61            60            61            60
baseline/execute/main/blake2b_huff/2805nulls_mean                           +0.0064         +0.0063          1004          1010          1004          1010
baseline/total/main/blake2b_huff/2805nulls_mean                             -0.0020         -0.0020          1022          1020          1022          1020
baseline/execute/main/blake2b_huff/5610nulls_mean                           +0.0095         +0.0095          1962          1981          1962          1981
baseline/total/main/blake2b_huff/5610nulls_mean                             +0.0009         +0.0009          1981          1982          1980          1982
baseline/execute/main/blake2b_huff/8415nulls_mean                           +0.0094         +0.0094          2875          2902          2875          2902
baseline/total/main/blake2b_huff/8415nulls_mean                             +0.0025         +0.0025          2904          2911          2904          2911
baseline/execute/main/blake2b_huff/65536nulls_mean                          +0.0086         +0.0086         22363         22556         22362         22555
baseline/total/main/blake2b_huff/65536nulls_mean                            -0.0035         -0.0035         22477         22399         22476         22398
baseline/analyse/main/blake2b_shifts_mean                                   +0.0016         +0.0016             9             9             9             9
baseline/execute/main/blake2b_shifts/2805nulls_mean                         +0.0099         +0.0099          8446          8530          8446          8530
baseline/total/main/blake2b_shifts/2805nulls_mean                           +0.0051         +0.0051          8471          8514          8471          8514
baseline/execute/main/blake2b_shifts/5610nulls_mean                         +0.0100         +0.0100         16845         17012         16844         17012
baseline/total/main/blake2b_shifts/5610nulls_mean                           +0.0062         +0.0062         16874         16979         16873         16978
baseline/execute/main/blake2b_shifts/8415nulls_mean                         +0.0102         +0.0102         25241         25498         25240         25497
baseline/total/main/blake2b_shifts/8415nulls_mean                           +0.0052         +0.0053         25285         25417         25283         25416
baseline/execute/main/blake2b_shifts/65536nulls_mean                        +0.0103         +0.0103        195649        197663        195639        197656
baseline/total/main/blake2b_shifts/65536nulls_mean                          +0.0041         +0.0041        196281        197082        196271        197074
baseline/analyse/main/sha1_divs_mean                                        +0.0046         +0.0046             1             1             1             1
baseline/execute/main/sha1_divs/empty_mean                                  +0.0042         +0.0042           147           148           147           148
baseline/total/main/sha1_divs/empty_mean                                    +0.0035         +0.0035           149           150           149           150
baseline/execute/main/sha1_divs/1351_mean                                   +0.0057         +0.0058          3102          3120          3102          3120
baseline/total/main/sha1_divs/1351_mean                                     +0.0051         +0.0051          3092          3107          3092          3107
baseline/execute/main/sha1_divs/2737_mean                                   +0.0068         +0.0068          6047          6088          6047          6088
baseline/total/main/sha1_divs/2737_mean                                     +0.0052         +0.0052          6026          6057          6026          6057
baseline/execute/main/sha1_divs/5311_mean                                   +0.0054         +0.0054         11814         11878         11814         11878
baseline/total/main/sha1_divs/5311_mean                                     +0.0048         +0.0048         11760         11817         11760         11816
baseline/execute/main/sha1_divs/65536_mean                                  +0.0062         +0.0062        143699        144596        143695        144592
baseline/total/main/sha1_divs/65536_mean                                    +0.0049         +0.0049        143144        143850        143140        143845
baseline/analyse/main/sha1_shifts_mean                                      -0.0136         -0.0136             1             1             1             1
baseline/execute/main/sha1_shifts/empty_mean                                -0.0011         -0.0011            95            95            95            95
baseline/total/main/sha1_shifts/empty_mean                                  +0.0021         +0.0020            98            98            98            98
baseline/execute/main/sha1_shifts/1351_mean                                 +0.0020         +0.0020          2029          2033          2029          2033
baseline/total/main/sha1_shifts/1351_mean                                   +0.0008         +0.0008          2034          2036          2034          2036
baseline/execute/main/sha1_shifts/2737_mean                                 +0.0021         +0.0021          3961          3970          3961          3970
baseline/total/main/sha1_shifts/2737_mean                                   +0.0002         +0.0002          3969          3969          3969          3969
baseline/execute/main/sha1_shifts/5311_mean                                 +0.0022         +0.0022          7734          7751          7733          7750
baseline/total/main/sha1_shifts/5311_mean                                   +0.0020         +0.0020          7743          7759          7743          7758
baseline/execute/main/sha1_shifts/65536_mean                                +0.0021         +0.0021         94450         94648         94446         94641
baseline/total/main/sha1_shifts/65536_mean                                  +0.0062         +0.0025         94310         94893         94306         94546
baseline/analyse/main/weierstrudel_mean                                     +0.0052         +0.0052            17            17            17            17
baseline/execute/main/weierstrudel/0_mean                                   +0.0039         +0.0039           485           487           485           487
baseline/total/main/weierstrudel/0_mean                                     +0.0029         +0.0031           504           506           504           506
baseline/execute/main/weierstrudel/1_mean                                   -0.0026         -0.0026          1008          1005          1008          1005
baseline/total/main/weierstrudel/1_mean                                     -0.0016         -0.0012          1022          1020          1021          1020
baseline/execute/main/weierstrudel/3_mean                                   -0.0052         -0.0050          1572          1564          1572          1564
baseline/total/main/weierstrudel/3_mean                                     -0.0020         -0.0018          1578          1575          1577          1574
baseline/execute/main/weierstrudel/9_mean                                   -0.0057         -0.0055          3236          3218          3236          3218
baseline/total/main/weierstrudel/9_mean                                     -0.0017         -0.0015          3214          3209          3213          3208
baseline/execute/main/weierstrudel/14_mean                                  -0.0052         -0.0051          4617          4593          4616          4593
baseline/total/main/weierstrudel/14_mean                                    -0.0025         -0.0023          4575          4563          4574          4563
baseline/analyse/micro/beginsub_push1s_0xffff_mean                          -0.0013         -0.0011           159           159           159           159
baseline/analyse/micro/beginsubs_0xffff_mean                                -0.0039         -0.0037            77            76            77            76
baseline/analyse/micro/jump_around_mean                                     +0.0010         +0.0012            33            33            33            33
baseline/execute/micro/jump_around/empty_mean                               -0.0014         -0.0013           113           113           113           113
baseline/total/micro/jump_around/empty_mean                                 +0.0008         +0.0009           149           149           149           149
baseline/analyse/micro/jumpdests_0xffff_mean                                -0.0009         -0.0007           184           184           184           184
baseline/execute/micro/jumpdests_0xffff/empty_mean                          -0.0009         -0.0008           185           185           185           185
baseline/total/micro/jumpdests_0xffff/empty_mean                            -0.0005         -0.0003           372           372           372           372
baseline/analyse/micro/loop_with_many_jumpdests_mean                        -0.0013         -0.0011            69            69            69            69
baseline/execute/micro/loop_with_many_jumpdests/empty_mean                  +0.0008         +0.0009         22387         22404         22382         22403
baseline/total/micro/loop_with_many_jumpdests/empty_mean                    -0.0031         -0.0029         22439         22370         22435         22369
baseline/analyse/micro/memory_grow_mload_mean                               +0.0000         +0.0002            24            24            24            24
baseline/execute/micro/memory_grow_mload/nogrow_mean                        +0.0020         +0.0033           120           120           120           120
baseline/total/micro/memory_grow_mload/nogrow_mean                          +0.0070         +0.0071           146           147           146           147
baseline/execute/micro/memory_grow_mload/by1_mean                           +0.0124         +0.0125           124           125           124           125
baseline/total/micro/memory_grow_mload/by1_mean                             +0.0025         +0.0027           150           151           150           151
baseline/execute/micro/memory_grow_mload/by16_mean                          +0.0086         +0.0088           138           139           138           139
baseline/total/micro/memory_grow_mload/by16_mean                            +0.0028         +0.0030           165           166           165           166
baseline/execute/micro/memory_grow_mload/by32_mean                          +0.0102         +0.0102           157           158           157           158
baseline/total/micro/memory_grow_mload/by32_mean                            +0.0032         +0.0032           185           185           185           185
baseline/analyse/micro/memory_grow_mstore_mean                              -0.0000         -0.0000            24            24            24            24
baseline/execute/micro/memory_grow_mstore/nogrow_mean                       -0.0009         -0.0009           135           135           135           135
baseline/total/micro/memory_grow_mstore/nogrow_mean                         -0.0019         -0.0019           162           161           162           161
baseline/execute/micro/memory_grow_mstore/by1_mean                          +0.0001         +0.0001           138           138           138           138
baseline/total/micro/memory_grow_mstore/by1_mean                            +0.0001         +0.0001           165           165           165           165
baseline/execute/micro/memory_grow_mstore/by16_mean                         -0.0008         -0.0008           154           154           154           154
baseline/total/micro/memory_grow_mstore/by16_mean                           +0.0017         +0.0017           181           182           181           182
baseline/execute/micro/memory_grow_mstore/by32_mean                         -0.0005         -0.0004           173           173           173           173
baseline/total/micro/memory_grow_mstore/by32_mean                           +0.0101         +0.0102           200           202           200           202
baseline/analyse/micro/push1s_0xffff_mean                                   -0.0008         -0.0008           168           168           168           168
baseline/analyse/micro/push32s_0xffff_mean                                  -0.0048         -0.0048            14            14            14            14
baseline/analyse/micro/signextend_mean                                      -0.0003         -0.0003            28            28            28            28
baseline/execute/micro/signextend/zero_mean                                 -0.0009         -0.0009           179           178           179           178
baseline/total/micro/signextend/zero_mean                                   +0.0002         +0.0002           209           209           209           209
baseline/execute/micro/signextend/one_mean                                  -0.0265         -0.0264           183           179           183           179
baseline/total/micro/signextend/one_mean                                    -0.0009         -0.0009           209           209           209           209
baseline/analyse/micro/zeros_0xffff_mean                                    -0.0004         -0.0004            76            76            76            76
baseline/total/synth/loop_v1_mean                                           +0.0049         +0.0049         14852         14924         14851         14923
baseline/total/synth/loop_v2_mean                                           +0.0024         +0.0024         17094         17135         17093         17135
baseline/total/synth/JUMPDEST/n0_mean                                       -0.0002         -0.0002          1487          1487          1487          1486
baseline/total/synth/ISZERO/u0_mean                                         -0.0004         -0.0004          2641          2640          2641          2640
baseline/total/synth/NOT/u0_mean                                            -0.0001         -0.0001          2349          2348          2349          2348
baseline/total/synth/ADD/b0_mean                                            -0.0015         -0.0015          3567          3562          3567          3562
baseline/total/synth/ADD/b1_mean                                            -0.0009         -0.0009          3924          3920          3924          3920
baseline/total/synth/MUL/b0_mean                                            -0.0000         -0.0000          7722          7722          7722          7722
baseline/total/synth/MUL/b1_mean                                            -0.0005         -0.0005          8044          8040          8044          8040
baseline/total/synth/SUB/b0_mean                                            -0.0004         -0.0004          4909          4907          4909          4907
baseline/total/synth/SUB/b1_mean                                            -0.0063         -0.0063          3437          3416          3437          3415
baseline/total/synth/SIGNEXTEND/b0_mean                                     +0.0003         +0.0004          2522          2523          2522          2523
baseline/total/synth/SIGNEXTEND/b1_mean                                     -0.0059         -0.0058          2528          2513          2528          2513
baseline/total/synth/LT/b0_mean                                             +0.0003         +0.0003          4092          4093          4091          4093
baseline/total/synth/LT/b1_mean                                             -0.0009         -0.0009          3671          3668          3671          3668
baseline/total/synth/GT/b0_mean                                             -0.0002         -0.0002          4091          4090          4091          4090
baseline/total/synth/GT/b1_mean                                             -0.0015         -0.0015          3664          3658          3664          3658
baseline/total/synth/SLT/b0_mean                                            -0.0012         -0.0011          4411          4406          4411          4406
baseline/total/synth/SLT/b1_mean                                            -0.0058         -0.0058          3988          3965          3988          3965
baseline/total/synth/SGT/b0_mean                                            -0.0001         -0.0001          4436          4436          4436          4436
baseline/total/synth/SGT/b1_mean                                            +0.0021         +0.0021          3964          3972          3963          3972
baseline/total/synth/EQ/b0_mean                                             +0.0001         +0.0001          4510          4510          4510          4510
baseline/total/synth/EQ/b1_mean                                             +0.0026         +0.0026          2803          2810          2803          2810
baseline/total/synth/AND/b0_mean                                            -0.0213         -0.0213          2652          2596          2652          2596
baseline/total/synth/AND/b1_mean                                            -0.0358         -0.0358          2768          2669          2768          2669
baseline/total/synth/OR/b0_mean                                             +0.0005         +0.0005          2559          2560          2558          2560
baseline/total/synth/OR/b1_mean                                             -0.0053         -0.0053          2652          2638          2652          2638
baseline/total/synth/XOR/b0_mean                                            +0.0000         -0.0000          2783          2783          2783          2783
baseline/total/synth/XOR/b1_mean                                            +0.0219         +0.0064          2791          2853          2791          2809
baseline/total/synth/BYTE/b0_mean                                           +0.0094         -0.0002          4677          4722          4677          4676
baseline/total/synth/BYTE/b1_mean                                           +0.0028         -0.0010          4478          4490          4478          4473
baseline/total/synth/SHL/b0_mean                                            +0.0013         -0.0005          4966          4972          4966          4963
baseline/total/synth/SHL/b1_mean                                            -0.0019         -0.0026          3826          3819          3826          3816
baseline/total/synth/SHR/b0_mean                                            -0.0023         -0.0023          4918          4906          4917          4906
baseline/total/synth/SHR/b1_mean                                            +0.0024         +0.0023          3897          3906          3897          3905
baseline/total/synth/SAR/b0_mean                                            +0.0006         +0.0003          6586          6590          6586          6588
baseline/total/synth/SAR/b1_mean                                            -0.0031         -0.0035          6731          6710          6731          6708
baseline/total/synth/ADDRESS/a0_mean                                        +0.0004         +0.0001          6273          6276          6273          6273
baseline/total/synth/ADDRESS/a1_mean                                        +0.0032         +0.0029          6573          6594          6573          6591
baseline/total/synth/CALLER/a0_mean                                         +0.0006         +0.0002          6273          6277          6273          6274
baseline/total/synth/CALLER/a1_mean                                         -0.0028         -0.0032          6592          6574          6592          6571
baseline/total/synth/CALLVALUE/a0_mean                                      +0.0005         +0.0004          2786          2787          2786          2787
baseline/total/synth/CALLVALUE/a1_mean                                      -0.0003         +0.0001          2794          2793          2794          2794
baseline/total/synth/CALLDATASIZE/a0_mean                                   -0.0009         -0.0006          2203          2201          2203          2201
baseline/total/synth/CALLDATASIZE/a1_mean                                   +0.0019         +0.0021          2211          2215          2211          2216
baseline/total/synth/CODESIZE/a0_mean                                       -0.0030         -0.0030          1962          1956          1962          1956
baseline/total/synth/CODESIZE/a1_mean                                       -0.0070         -0.0070          2107          2092          2106          2092
baseline/total/synth/RETURNDATASIZE/a0_mean                                 -0.0000         -0.0001          1894          1894          1894          1894
baseline/total/synth/RETURNDATASIZE/a1_mean                                 +0.0011         +0.0010          2109          2111          2109          2111
baseline/total/synth/PC/a0_mean                                             +0.0000         -0.0001          2202          2202          2202          2202
baseline/total/synth/PC/a1_mean                                             -0.0056         -0.0058          2229          2216          2229          2216
baseline/total/synth/MSIZE/a0_mean                                          -0.0008         -0.0010          2206          2204          2205          2203
baseline/total/synth/MSIZE/a1_mean                                          +0.0011         +0.0009          2210          2213          2210          2212
baseline/total/synth/GAS/a0_mean                                            +0.0001         -0.0001          2207          2208          2207          2207
baseline/total/synth/GAS/a1_mean                                            +0.0044         +0.0042          2217          2227          2217          2226
baseline/total/synth/PUSH1/p0_mean                                          +0.0007         +0.0004          4668          4671          4667          4669
baseline/total/synth/PUSH1/p1_mean                                          +0.0096         +0.0093          4857          4903          4856          4902
baseline/total/synth/PUSH2/p0_mean                                          +0.0003         +0.0000          4532          4533          4532          4532
baseline/total/synth/PUSH2/p1_mean                                          +0.0015         +0.0012          4759          4766          4759          4765
baseline/total/synth/PUSH3/p0_mean                                          +0.0003         +0.0007          4823          4825          4823          4826
baseline/total/synth/PUSH3/p1_mean                                          +0.0029         +0.0032          5060          5075          5060          5076
baseline/total/synth/PUSH4/p0_mean                                          +0.0000         +0.0003          4242          4242          4241          4242
baseline/total/synth/PUSH4/p1_mean                                          +0.0008         +0.0010          4410          4413          4410          4414
baseline/total/synth/PUSH5/p0_mean                                          -0.0004         -0.0002          4820          4818          4819          4818
baseline/total/synth/PUSH5/p1_mean                                          +0.0037         +0.0038          5051          5070          5051          5070
baseline/total/synth/PUSH6/p0_mean                                          -0.0000         +0.0000          4678          4678          4678          4678
baseline/total/synth/PUSH6/p1_mean                                          +0.0017         +0.0017          4847          4855          4846          4855
baseline/total/synth/PUSH7/p0_mean                                          +0.0004         +0.0004          4969          4971          4968          4970
baseline/total/synth/PUSH7/p1_mean                                          +0.0003         +0.0003          5207          5209          5207          5209
baseline/total/synth/PUSH8/p0_mean                                          +0.0021         +0.0020          2219          2223          2219          2223
baseline/total/synth/PUSH8/p1_mean                                          -0.0013         -0.0014          2218          2216          2218          2215
baseline/total/synth/PUSH9/p0_mean                                          -0.0001         -0.0002          6713          6712          6713          6711
baseline/total/synth/PUSH9/p1_mean                                          -0.0028         -0.0029          6896          6877          6895          6876
baseline/total/synth/PUSH10/p0_mean                                         +0.0001         +0.0000          6568          6569          6568          6568
baseline/total/synth/PUSH10/p1_mean                                         -0.0001         -0.0003          6745          6744          6745          6743
baseline/total/synth/PUSH11/p0_mean                                         +0.0013         +0.0012          6861          6870          6861          6869
baseline/total/synth/PUSH11/p1_mean                                         +0.0012         +0.0010          7053          7061          7053          7060
baseline/total/synth/PUSH12/p0_mean                                         +0.0004         +0.0002          6423          6425          6423          6424
baseline/total/synth/PUSH12/p1_mean                                         +0.0001         -0.0000          6589          6590          6589          6589
baseline/total/synth/PUSH13/p0_mean                                         -0.0002         -0.0003          6861          6860          6861          6859
baseline/total/synth/PUSH13/p1_mean                                         +0.0014         +0.0012          7044          7054          7044          7053
baseline/total/synth/PUSH14/p0_mean                                         -0.0003         -0.0005          6737          6735          6737          6733
baseline/total/synth/PUSH14/p1_mean                                         -0.0006         -0.0007          6904          6900          6904          6898
baseline/total/synth/PUSH15/p0_mean                                         +0.0010         +0.0008          7006          7013          7005          7011
baseline/total/synth/PUSH15/p1_mean                                         -0.0005         -0.0007          7196          7192          7195          7191
baseline/total/synth/PUSH16/p0_mean                                         +0.0001         -0.0001          2354          2354          2354          2354
baseline/total/synth/PUSH16/p1_mean                                         +0.0005         +0.0003          2363          2364          2363          2363
baseline/total/synth/PUSH17/p0_mean                                         +0.0004         +0.0002          8314          8317          8313          8315
baseline/total/synth/PUSH17/p1_mean                                         +0.0033         +0.0031          8480          8508          8479          8506
baseline/total/synth/PUSH18/p0_mean                                         +0.0004         +0.0002          8169          8172          8169          8170
baseline/total/synth/PUSH18/p1_mean                                         +0.0015         +0.0013          8336          8349          8336          8347
baseline/total/synth/PUSH19/p0_mean                                         +0.0002         +0.0000          8325          8326          8324          8324
baseline/total/synth/PUSH19/p1_mean                                         +0.0006         +0.0003          8532          8537          8531          8534
baseline/total/synth/PUSH20/p0_mean                                         +0.0008         +0.0004          8028          8035          8028          8031
baseline/total/synth/PUSH20/p1_mean                                         +0.0002         +0.0002          8211          8212          8211          8212
baseline/total/synth/PUSH21/p0_mean                                         -0.0001         +0.0005          8176          8175          8176          8179
baseline/total/synth/PUSH21/p1_mean                                         -0.0000         +0.0005          8356          8356          8356          8360
baseline/total/synth/PUSH22/p0_mean                                         -0.0004         +0.0000          8177          8174          8176          8176
baseline/total/synth/PUSH22/p1_mean                                         +0.0015         +0.0018          8371          8383          8370          8385
baseline/total/synth/PUSH23/p0_mean                                         +0.0001         +0.0004          8466          8467          8465          8469
baseline/total/synth/PUSH23/p1_mean                                         +0.0010         +0.0012          8657          8666          8657          8667
baseline/total/synth/PUSH24/p0_mean                                         -0.2201         -0.0286          2739          2136          2199          2136
baseline/total/synth/PUSH24/p1_mean                                         -0.0222         -0.0218          2605          2547          2604          2547
baseline/total/synth/PUSH25/p0_mean                                         +0.0001         +0.0001          8616          8616          8615          8616
baseline/total/synth/PUSH25/p1_mean                                         +0.0002         +0.0003          8783          8785          8783          8785
baseline/total/synth/PUSH26/p0_mean                                         -0.0007         -0.0008          8467          8461          8467          8460
baseline/total/synth/PUSH26/p1_mean                                         -0.0004         -0.0004          8680          8677          8680          8677
baseline/total/synth/PUSH27/p0_mean                                         -0.0007         -0.0010          8767          8760          8766          8757
baseline/total/synth/PUSH27/p1_mean                                         +0.0023         +0.0023          8923          8943          8923          8943
baseline/total/synth/PUSH28/p0_mean                                         +0.0001         +0.0001          8320          8321          8320          8321
baseline/total/synth/PUSH28/p1_mean                                         +0.0008         +0.0008          8491          8498          8491          8498
baseline/total/synth/PUSH29/p0_mean                                         +0.0002         +0.0002          8753          8755          8753          8755
baseline/total/synth/PUSH29/p1_mean                                         +0.0004         +0.0004          8919          8923          8919          8922
baseline/total/synth/PUSH30/p0_mean                                         -0.0000         -0.0001          6884          6883          6883          6883
baseline/total/synth/PUSH30/p1_mean                                         +0.0007         +0.0006          7029          7034          7029          7034
baseline/total/synth/PUSH31/p0_mean                                         -0.0000         -0.0000          8761          8761          8761          8760
baseline/total/synth/PUSH31/p1_mean                                         -0.0000         -0.0001          8941          8941          8941          8940
baseline/total/synth/PUSH32/p0_mean                                         +0.0021         +0.0021          2277          2282          2277          2282
baseline/total/synth/PUSH32/p1_mean                                         -0.0011         -0.0011          2561          2559          2561          2558
baseline/total/synth/SWAP1/s0_mean                                          -0.0002         -0.0002          2937          2936          2937          2936
baseline/total/synth/SWAP2/s0_mean                                          +0.0006         +0.0005          2934          2936          2934          2936
baseline/total/synth/SWAP3/s0_mean                                          -0.0001         -0.0001          2935          2935          2935          2935
baseline/total/synth/SWAP4/s0_mean                                          +0.0070         +0.0069          2938          2958          2937          2958
baseline/total/synth/SWAP5/s0_mean                                          -0.0022         -0.0022          3236          3229          3236          3229
baseline/total/synth/SWAP6/s0_mean                                          -0.0001         -0.0002          2943          2942          2943          2942
baseline/total/synth/SWAP7/s0_mean                                          +0.0002         +0.0001          3233          3234          3233          3233
baseline/total/synth/SWAP8/s0_mean                                          -0.0011         -0.0012          2684          2681          2684          2680
baseline/total/synth/SWAP9/s0_mean                                          +0.0003         +0.0002          3237          3238          3237          3237
baseline/total/synth/SWAP10/s0_mean                                         -0.0016         -0.0017          2718          2714          2718          2714
baseline/total/synth/SWAP11/s0_mean                                         -0.0010         -0.0011          2953          2950          2953          2950
baseline/total/synth/SWAP12/s0_mean                                         -0.0003         -0.0002          2953          2952          2953          2952
baseline/total/synth/SWAP13/s0_mean                                         +0.0052         +0.0053          2690          2704          2690          2705
baseline/total/synth/SWAP14/s0_mean                                         -0.0003         -0.0002          2957          2956          2957          2956
baseline/total/synth/SWAP15/s0_mean                                         -0.0010         -0.0009          2706          2703          2705          2703
baseline/total/synth/SWAP16/s0_mean                                         -0.0067         -0.0066          2995          2975          2995          2975
baseline/total/synth/DUP1/d0_mean                                           -0.0007         -0.0007          2062          2061          2062          2061
baseline/total/synth/DUP1/d1_mean                                           +0.0076         +0.0077          2074          2090          2074          2090
baseline/total/synth/DUP2/d0_mean                                           +0.0036         +0.0037          2201          2209          2201          2209
baseline/total/synth/DUP2/d1_mean                                           +0.0001         +0.0001          2212          2212          2212          2212
baseline/total/synth/DUP3/d0_mean                                           +0.0077         +0.0078          2043          2059          2043          2059
baseline/total/synth/DUP3/d1_mean                                           +0.0010         +0.0011          2075          2077          2075          2077
baseline/total/synth/DUP4/d0_mean                                           +0.0003         +0.0003          2060          2060          2060          2060
baseline/total/synth/DUP4/d1_mean                                           -0.0016         -0.0015          2073          2070          2073          2070
baseline/total/synth/DUP5/d0_mean                                           -0.0004         -0.0004          2346          2345          2346          2345
baseline/total/synth/DUP5/d1_mean                                           -0.0044         -0.0044          2369          2359          2369          2359
baseline/total/synth/DUP6/d0_mean                                           +0.0041         +0.0041          2193          2202          2193          2202
baseline/total/synth/DUP6/d1_mean                                           +0.0011         +0.0011          2207          2209          2207          2209
baseline/total/synth/DUP7/d0_mean                                           -0.0067         -0.0067          2071          2057          2071          2057
baseline/total/synth/DUP7/d1_mean                                           -0.0011         -0.0011          2083          2080          2082          2080
baseline/total/synth/DUP8/d0_mean                                           +0.0006         +0.0006          2201          2202          2201          2202
baseline/total/synth/DUP8/d1_mean                                           +0.0009         +0.0008          2211          2213          2211          2213
baseline/total/synth/DUP9/d0_mean                                           -0.0047         -0.0047          2212          2201          2212          2201
baseline/total/synth/DUP9/d1_mean                                           +0.0032         +0.0032          2216          2223          2216          2223
baseline/total/synth/DUP10/d0_mean                                          -0.0120         -0.0121          2230          2203          2230          2203
baseline/total/synth/DUP10/d1_mean                                          -0.0243         -0.0244          2273          2218          2273          2218
baseline/total/synth/DUP11/d0_mean                                          -0.0050         -0.0050          2214          2203          2214          2202
baseline/total/synth/DUP11/d1_mean                                          -0.0013         -0.0014          2220          2217          2220          2217
baseline/total/synth/DUP12/d0_mean                                          +0.0013         +0.0013          2201          2204          2201          2203
baseline/total/synth/DUP12/d1_mean                                          +0.0019         +0.0018          2216          2220          2216          2220
baseline/total/synth/DUP13/d0_mean                                          +0.0026         +0.0026          2343          2349          2343          2349
baseline/total/synth/DUP13/d1_mean                                          +0.0017         +0.0016          2353          2357          2353          2357
baseline/total/synth/DUP14/d0_mean                                          +0.0032         +0.0032          2191          2198          2190          2197
baseline/total/synth/DUP14/d1_mean                                          +0.0034         +0.0033          2204          2211          2203          2211
baseline/total/synth/DUP15/d0_mean                                          +0.0024         +0.0024          2061          2066          2061          2066
baseline/total/synth/DUP15/d1_mean                                          -0.0006         -0.0007          2086          2085          2086          2084
baseline/total/synth/DUP16/d0_mean                                          -0.0015         -0.0016          2209          2206          2209          2205
baseline/total/synth/DUP16/d1_mean              
```

File: lib/evmone/baseline.cpp [hunk 1]
```diff
@@ -47,36 +47,40 @@ CodeAnalysis analyze(const uint8_t* code, size_t code_size)
...
-    if (INTX_UNLIKELY((state.gas_left -= metrics.gas_cost) < 0))
+    if (INTX_UNLIKELY((state.gas_left -= cost_table[Op]) < 0))
         return EVMC_OUT_OF_GAS;
 
     const auto stack_size = state.stack.size();
-    if (INTX_UNLIKELY(stack_size == Stack::limit))
+    if constexpr (instr::traits[Op].stack_height_change > 0)
     {
-        if (metrics.can_overflow_stack)
+        if (INTX_UNLIKELY(stack_size == Stack::limit))
```
[the remark this answers is not in the document]
Comment (chfast) [reply lib/evmone/baseline.cpp:64]: Done. thanks.

File: lib/evmone/baseline.cpp [hunk 2]
```diff
@@ -136,6 +139,19 @@ template <>
     return instr_fn(state, pos);
 }
 
+/// A helper to invoke the instruction implementation of the given opcode Op.
+template <evmc_opcode Op>
+[[gnu::always_inline]] inline code_iterator invoke(
```
Comment (gumb0) [inline lib/evmone/baseline.cpp:144]: Still I wonder if it's simpler to inline this helper manually into the macro.

**Review by gumb0 — APPROVED** [review]

File: lib/evmone/baseline.cpp (hunk 2 above)
Comment (chfast) [reply lib/evmone/baseline.cpp:144]: I want to have the macro as small as possible: it is only needed for `goto exit`.

**Pull request merged**